=== modified file 'libgearman/worker.cc'
--- libgearman/worker.cc	2013-11-07 09:30:17 +0000
+++ libgearman/worker.cc	2014-01-27 22:59:21 +0000
@@ -830,7 +830,7 @@
 
                 if (gearman_failed(*ret_ptr))
                 {
-                  if (*ret_ptr == GEARMAN_IO_WAIT)
+                  if (*ret_ptr == GEARMAN_IO_WAIT || *ret_ptr == GEARMAN_TIMEOUT)
                   {
                     worker->state= GEARMAN_WORKER_STATE_GRAB_JOB_RECV;
                   }

